# BEE 4750 Homework 5: Solid Waste Disposal

**Name**: Jonathan Marcuse

**ID**: jrm564

> **Due Date**
>
> Friday, 11/10/23, 9:00pm

## Overview

### Instructions

-   In Problem 1, you will formulate, solve, and analyze a standard
    generating capacity expansion problem.
-   In Problem 2, you will add a CO<sub>2</sub> constraint to the
    capacity expansion problem and identify changes in the resulting
    solution.

### Load Environment

The following code loads the environment and makes sure all needed
packages are installed. This should be at the start of most Julia
scripts.

``` julia
import Pkg
Pkg.activate(@__DIR__)
Pkg.instantiate()
```

``` julia
using JuMP
using HiGHS
using DataFrames
using GraphRecipes
using Plots
using Measures
using MarkdownTables
```

In [1]:
import Pkg
Pkg.activate(@__DIR__)
Pkg.instantiate()

  Activating project at `~/Desktop/BEE4750-2/hw05-jrmarcuse`


   Installed LoggingExtras ────── v1.0.3


   Installed CodecBzip2 ───────── v0.8.1


   Installed HiGHS_jll ────────── v1.6.0+0


   Installed fzf_jll ──────────── v0.35.1+0
   Installed TranscodingStreams ─ v0.10.1


   Installed NetworkLayout ────── v0.4.6


   Installed JLFzf ────────────── v0.1.6
   Installed StaticArrays ─────── v1.6.5


   Installed Extents ──────────── v0.1.2
   Installed Inflate ──────────── v0.1.4
   Installed MathOptInterface ─── v1.21.0


   Installed Graphs ───────────── v1.9.0


   Installed CodecZlib ────────── v0.7.3


   Installed ConstructionBase ─── v1.5.4


   Installed Adapt ────────────── v3.7.1
   Installed HiGHS ────────────── v1.7.5
   Installed ChainRulesCore ───── v1.18.0
   Installed XML2_jll ─────────── v2.11.5+0
   Installed JuMP ─────────────── v1.16.0


   Installed URIs ─────────────── v1.5.1
   Installed StructArrays ─────── v0.6.16


   Installed GeometryBasics ───── v0.4.9


   Installed Tables ───────────── v1.11.1


   Installed GeoInterface ─────── v1.3.2


   Installed RelocatableFolders ─ v1.0.1


   Installed SortingAlgorithms ── v1.2.0


   Installed PrettyTables ─────── v2.2.8


Precompiling 

project...


  ✓ Extents


  ✓ URIs


  ✓ Inflate


  ✓ TranscodingStreams


  ✓ ConstructionBase


  ✓ Adapt


  ✓ LoggingExtras


  ✓ ChainRulesCore


  ✓ SimpleTraits


  ✓ RelocatableFolders


  ✓ Tables


  ✓ fzf_jll


  ✓ HiGHS_jll


  ✓ EarCut_jll


  ✓ SortingAlgorithms


  ✓ XML2_jll


  ✓ TranscodingStreams → TestExt


  ✓ CodecZlib


  ✓ GeoInterface


  ✓ ChainRulesCore → ChainRulesCoreSparseArraysExt


  ✓ Unitful → ConstructionBaseUnitfulExt


  ✓ MarkdownTables


  ✓ JLFzf
  ✓ LogExpFunctions → LogExpFunctionsChainRulesCoreExt


  ✓ XSLT_jll


  ✓ Wayland_jll


  ✓ Gettext_jll


  ✓ CodecBzip2


  ✓ StatsBase


  ✓ Xorg_libxcb_jll


  ✓ SpecialFunctions


  ✓ StaticArrays


  ✓ Xorg_xcb_util_jll


  ✓ HTTP


  ✓ Glib_jll


  ✓ Xorg_libX11_jll


  ✓ DiffRules


  ✓ ColorVectorSpace → SpecialFunctionsExt


  ✓ SpecialFunctions → SpecialFunctionsChainRulesCoreExt


  ✓ ArnoldiMethod


  ✓ StaticArrays → StaticArraysStatisticsExt


  ✓ ConstructionBase → ConstructionBaseStaticArraysExt


  ✓ Xorg_xcb_util_image_jll


  ✓ Adapt → AdaptStaticArraysExt


  ✓ Xorg_xcb_util_keysyms_jll


  ✓ Xorg_xcb_util_renderutil_jll


  ✓ Xorg_xcb_util_wm_jll


  ✓ Xorg_libXrender_jll
  ✓ Xorg_libXext_jll


  ✓ Xorg_libXfixes_jll


  ✓ Xorg_libxkbfile_jll


  ✓ GeometryTypes


  ✓ OffsetArrays


  ✓ GPUArraysCore


  ✓ Xorg_xcb_util_cursor_jll


  ✓ Libglvnd_jll


  ✓ ForwardDiff


  ✓ Xorg_libXinerama_jll


  ✓ Xorg_libXrandr_jll


  ✓ Xorg_libXcursor_jll


  ✓ Graphs


  ✓ PrettyTables


  ✓ Cairo_jll


  ✓ Xorg_libXi_jll


  ✓ Xorg_xkbcomp_jll


  ✓ StructArrays


  ✓ ForwardDiff → ForwardDiffStaticArraysExt


  ✓ ColorSchemes


  ✓ HarfBuzz_jll


  ✓ Interpolations
  ✓ GLFW_jll


  ✓ Xorg_xkeyboard_config_jll


  ✓ libass_jll


  ✓ xkbcommon_jll


  ✓ FFMPEG_jll


  ✓ Vulkan_Loader_jll


  ✓ FFMPEG


  ✓ Qt6Base_jll


  ✓ GeometryBasics


  ✓ GR_jll


  ✓ NetworkLayout


  ✓ NetworkLayout → NetworkLayoutGraphsExt


  ✓ GR


  ✓ PlotUtils


  ✓ PlotThemes


  ✓ GraphRecipes


  ✓ RecipesPipeline


  ✓ DataFrames


  ✓ Latexify → DataFramesExt


  ✓ MathOptInterface


  ✓ Plots


  ✓ HiGHS


  ✓ Plots → UnitfulExt


  ✓ Plots → GeometryBasicsExt


  ✓ JuMP


  95 dependencies successfully precompiled in 2929 seconds. 120 already precompiled.


In [2]:
using JuMP
using HiGHS
using DataFrames
using GraphRecipes
using Plots
using Measures
using MarkdownTables

## Background

Three cities are developing a coordinated municipal solid waste (MSW)
disposal plan. Three disposal alternatives are being considered: a
landfill (LF), a materials recycling facility (MRF), and a
waste-to-energy facility (WTE). The capacities of these facilities and
the fees for operation and disposal are provided in the table below.

|    **Disposal Facility**     | **Capacity** (Mg/d) | **Fixed cost** (\$/d) | **Tipping Fee ** (\$/Mg) | **Recycling Cost** (\$/Mg) |
|:-------------:|:-------------:|:-------------:|:----------:|:--------------:|
|           Landfill           |         200         |         2000          |            50            |                            |
| Materials Recycling Facility |         350         |         1500          |            7             |    40 (per Mg recycled)    |
|   Waste-to-Energy Facility   |         210         |         2500          |            60            |                            |

Transportation costs are \$1.5/Mg-km, and the relative distances between
the cities and facilities are provided in the table below.

| **City/Facility** | **Landfill (km)** | **MRF (km)** | **WTE (km)** |
|:-----------------:|:-----------------:|:------------:|:------------:|
|         1         |         5         |      30      |      15      |
|         2         |        15         |      25      |      10      |
|         3         |        13         |      45      |      20      |
|        LF         |        \-         |      32      |      18      |
|        MRF        |        32         |      \-      |      15      |
|        WTE        |        18         |      15      |      \-      |

The fixed costs associated with the disposal options are incurred only
if the particular disposal option is implemented. The three cities
produce 100, 90, and 120 Mg/day of solid waste, respectively, with the
composition provided in the table below.

|   **Component**   | **% of total mass** | **Combustion ash** (%) | **MRF Recycling rate** (%) |
|:---------------------:|:--------------:|:---------------:|:---------------:|
|    Food Wastes    |         15          |           8            |             0              |
| Paper & Cardboard |         40          |           7            |             55             |
|     Plastics      |          5          |           5            |             15             |
|     Textiles      |          3          |           10           |             10             |
|  Rubber, Leather  |          2          |           15           |             0              |
|       Wood        |          5          |           2            |             30             |
|    Yard Wastes    |         18          |           2            |             40             |
|       Glass       |          4          |          100           |             60             |
|      Ferrous      |          2          |          100           |             75             |
|     Aluminum      |          2          |          100           |             80             |
|    Other Metal    |          1          |          100           |             50             |
|   Miscellaneous   |          3          |           70           |             0              |

The information in the above table will help you determine the overall
recycling and ash fractions. Note that the recycling residuals, which
may be sent to either landfill or the WTE, have different ash content
than the ash content of the original MSW. You will need to determine
these fractions to construct your mass balance constraints.

**Reminder**: Use `round(x; digits=n)` to report values to the
appropriate precision!

## Problems (Total: 40 Points)

### Problem 1 (22 points)

In this problem, you will develop an optimal disposal plan for the two
cities.

#### Problem 1.1 (3 points)

Based on the information above, calculate the overall recycling and ash
fractions for the waste produced by each city.

#### Problem 1.2 (2 points)

What are the decision variables for your optimization problem? Provide
notation and variable meaning.

#### Problem 1.3 (3 points)

Formulate the objective function. Make sure to include any needed
derivations or justifications for your equation(s).

#### Problem 1.4 (4 points)

Derive all relevant constraints. Make sure to include any needed
justifications or derivations.

#### Problem 1.5 (3 points)

Implement your optimization problem in `JuMP`.

#### Problem 1.6 (2 points)

Find the optimal solution. Report the optimal objective value.

#### Problem 1.7 (5 points)

Draw a diagram showing the flows of waste between the cities and the
facilities. Which facilities (if any) will not be used? Does this
solution make sense?

### Problem 2 (18 points)

It is projected that in the near future the state will introduce a
carbon tax that will increase the cost for transportation and for
disposal by incineration. It is estimated that the additional costs will
be:

-   tipping fee for the WTE facility will increase to\$75/Mg; and
-   transportation costs will increase to \$2/Mg-km.

In this context, the cities are considering adding another landfill and
want to know if this would be cost-effective compared to using the
current facilities with the carbon tax. This landfill would have a
maximum capacity of 100 Mg/day and would be located with the following
distances from the existing sites (excluding LF1):

| **City/Facility** | **Distance to LF2 (km)** |
|:-----------------:|:------------------------:|
|         1         |            45            |
|         2         |            35            |
|         3         |            15            |
|        MRF        |            35            |
|        WTE        |            50            |

The fixed cost of operating this facility would be the same as the first
landfill, but the tipping cost would be increased to \$60/Mg-day.

#### Problem 2.1 (5 points)

What changes are needed to your optimization program from Problem 1 for
this decision problem? Formulate any different variables, objectives,
and/or constraints.

#### Problem 2.2 (3 points)

Implement the new optimization problem in `JuMP`.

#### Problem 2.3 (5 points)

Find the optimal solution and report the optimal objective value.
Provide a diagram showing the new waste flows.

#### Problem 2.4 (5 points)

Would you recommend that the cities build the new landfill? Why or why
not? Your answer should be based on your analysis but can draw on other
considerations as appropriate or desired.

## References

List any external references consulted, including classmates.